source/app/models/url.rb

```diff
@@ -0,0 +1,18 @@
+class Url < ActiveRecord::Base
+  validates :original, presence: true
+  validates :original, length: { minimum: 1 }
+  validates :original, format: {with: /\A((http|https):\/\/).*\Z/}
```

Comment:
The `.*\Z` part is unneeded. All strings will have some more content and eventually end. You don't need to specify it.